tck/src/main/java/org/eclipse/microprofile/lra/tck/service/LRATestService.java

```diff
@@ -105,4 +115,83 @@ public void waitForEndPhaseReplay(URI lraId) {
 
         return iterator.next();
     }
+
+    public void assertHeaderPresent(URI lraId, String path, String headerName) {
+        if (lraId == null) {
+            throw new WrongHeaderException(String.format("%s: missing '%s' header", path, headerName));
+        }
+    }
+
+    public Response processAfterLRAInfo(URI endedLRAId, LRAStatus status, String resourceName, String path) {
+        assertHeaderPresent(endedLRAId, path, LRA_HTTP_ENDED_CONTEXT_HEADER);
+
+        switch (status) {
+            case Closed:
+                // FALLTHRU
+            case Cancelled:
+                // FALLTHRU
+            case FailedToCancel:
+                // FALLTHRU
+            case FailedToClose:
+                if (resourceName != null) {
+                    lraMetricService.incrementMetric(
+                        LRAMetricType.valueOf(status.name()),
+                        endedLRAId,
+                        resourceName);
+                } else {
+                    lraMetricService.incrementMetric(
+                        LRAMetricType.valueOf(status.name()),
+                        endedLRAId);
+                }
+                return Response.ok().build();
+            default:
+                return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+
+    }
+
+    /**
+     * Returns whether the passed LRA and resource are in finished state.
+     *
+     * @param lra the LRA to test
+     * @param resourceName name of the resource that the metrics parameter applies to
+     * @return whether or not an LRA has finished
+     */
+    public boolean isLRAFinished(URI lra, String resourceName) {
+        return lraMetricService.getMetric(LRAMetricType.Closed, lra, resourceName) > 0 ||
+            lraMetricService.getMetric(LRAMetricType.FailedToClose, lra, resourceName) > 0 ||
+            lraMetricService.getMetric(LRAMetricType.Cancelled, lra, resourceName) > 0 ||
+            lraMetricService.getMetric(LRAMetricType.FailedToCancel, lra, resourceName) > 0;
+    }
+
+    /**
+     * Returns whether the passed LRA is in finished state.
+     *
+     * @param lra the LRA to test
+     * @return whether or not an LRA has finished
+     */
+    public boolean isLRAFinished(URI lra) {
+        return isLRAFinished(lra, LRAMetricService.ALL);
+    }
+
+    /**
+     * Returns the end {@link LRAStatus} of the passed LRA or null if the LRA is not in end state.
+     *
+     * @param lra the LRA whose end status is being queried
+     * @param metricFunction the metric function to be used to get the metrics
+     * @return the end status of the LRA (or null if the LRA has not yet finished)
+     */
+    private LRAStatus getLRAEndStatus(URI lra, Function<LRAMetricType, Integer> metricFunction) {
```

Comment:
This method is no longer used. Otherwise the PR looks good

Reply:
@xstefank When you have resolved this comment we can look at getting this (and the remaining PRs merged). That will allow us to produce a release of milestone 1.

Reply:
sorry, PTO. On it.